pkg/sqlmigrations/migrations.go

```diff
@@ -1020,9 +1020,27 @@ func migrateMutationJobForTable(
 			continue
 		}
 		log.VEventf(
-			ctx, 2, "job %d: found corresponding mutation %d on table %d",
+			ctx, 2, "job %d: found corresponding MutationJob %d on table %d",
 			*job.ID(), mutationJob.MutationID, tableDesc.ID,
 		)
+		var mutation *sqlbase.DescriptorMutation
+		found := false
+		for i := range tableDesc.Mutations {
+			if tableDesc.Mutations[i].MutationID == mutationJob.MutationID {
+				mutation = &tableDesc.Mutations[i]
+				found = true
+				break
+			}
+		}
+		if !found {
```

Comment:
Does it make sense to check if `mutation` is nil rather than maintaining the `found` flag? I think this would provide a stronger guarantee.